jcl/src/openj9.dtfj/share/classes/com/ibm/dtfj/corereaders/NewElfDump.java

```diff
@@ -469,8 +470,44 @@ public Address getLinkRegisterFrom(Map registers) {
 		public long readUID(ElfFile file) throws IOException {
 			return file.readInt() & 0xffffffffL;
 		}
-    }
-    
+	}
+
+	private static class ArchAARCH64 implements Arch {
+		public Map readRegisters(ElfFile file, Builder builder, Object addressSpace) throws IOException {
+			Map registers = new TreeMap();
+			for (int i = 0; i < 30; i++) {
+				registers.put( "x"+i, new Address64(file.readLong()) );
+			}
+			// The next registers have names
+			registers.put( "lr", new Address64(file.readLong()) );
+			registers.put( "sp", new Address64(file.readLong()) );
+			registers.put( "pc", new Address64(file.readLong()) );
+			registers.put( "pstate", new Address64(file.readLong()) );
+
+			return registers;
+		}
+
+		public Address getStackPointerFrom(Map registers) {
+			return (Address) registers.get("sp");
+		}
+
+		public Address getBasePointerFrom(Map registers) {
+			return getStackPointerFrom(registers);
+		}
+
+		public Address getInstructionPointerFrom(Map registers) {
+			return (Address) registers.get("pc");
+		}
+
+		public Address getLinkRegisterFrom(Map registers) {
+			return (Address) registers.get("lr");
+		}
+
+		public long readUID(ElfFile file) throws IOException {
+			return file.readInt() & 0xffffffffL;
```

Comment:
This seems inconsistent when compared to `ELFAArch64DumpReader.readUID()` - please explain why they should not read a value of the same width.

Reply:
Thank you for pointing that out.
It seems I wrote the `ELFAArch64DumpReader` code based on `ELFARM32DumpReader`, and I didn't notice the inconsistency at that time.